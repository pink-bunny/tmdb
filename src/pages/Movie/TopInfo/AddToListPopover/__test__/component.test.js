import React from 'react';
import { shallow } from 'enzyme';

import AddToListComponent from '../component';

describe('AddToList component', () => {
  const requiredProps = {
    addToList: jest.fn(),
    closePopover: jest.fn(),
    popoverVisible: true,
    handleVisiblePopover: jest.fn(),
    popoverCreateListModalSubmit: jest.fn()
  };
  const wrapper = shallow(<AddToListComponent {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with optional props', () => {
    wrapper.setProps({
      list: [{ id: 1, name: 'Mock name' }]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
