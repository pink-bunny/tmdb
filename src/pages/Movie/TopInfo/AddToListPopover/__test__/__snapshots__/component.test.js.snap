// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToList component matches snapshot with optional props 1`] = `
<Popover
  content={
    <PopoverContent
      addToList={[MockFunction]}
      closePopover={[MockFunction]}
      createListModalSubmit={[MockFunction]}
      list={
        Array [
          Object {
            "id": 1,
            "name": "Mock name",
          },
        ]
      }
    />
  }
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  onVisibleChange={[MockFunction]}
  overlayClassName="detailed-movie__add-to-list-popover"
  overlayStyle={Object {}}
  placement="top"
  title="Add movie to list"
  transitionName="zoom-big"
  trigger="click"
  visible={true}
>
  <Icon
    type="plus-circle"
  />
</Popover>
`;

exports[`AddToList component matches snapshot with required props 1`] = `
<Popover
  content={
    <PopoverContent
      addToList={[MockFunction]}
      closePopover={[MockFunction]}
      createListModalSubmit={[MockFunction]}
      list={null}
    />
  }
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  onVisibleChange={[MockFunction]}
  overlayClassName="detailed-movie__add-to-list-popover"
  overlayStyle={Object {}}
  placement="top"
  title="Add movie to list"
  transitionName="zoom-big"
  trigger="click"
  visible={true}
>
  <Icon
    type="plus-circle"
  />
</Popover>
`;
